services/gossip/dispatcher.go

```diff
@@ -78,14 +85,15 @@ func (c *meteredTopicChannel) flushChannelAfterContextDone(expiredContext contex
 }
 
 func newMeteredTopicChannel(name string, registry metric.Registry) *meteredTopicChannel {
-	bufferSize := 100 // TODO - when this is reduced TestLeanHelix_RecoversFromDiskError fails - whats the correct buffer size?
+	bufferSize := 10 // TODO increase significantly once main peer socket size is reduced
```

Comment:
peer socket size is only for `memoryTransport`... what's the equivalent for TCP?

Reply:
The flow is a little different in direct transport: the client maintains an outgoing queue as a buffered channel. the buffer size is `const SEND_QUEUE_MAX_MESSAGES = 1000` when a client tries to send a message to a peer it first writes to this buffer _unless it's full_. if it's full the message is discarded and the error is more or less ignored.

after writing to the outgoing queue buffer, the client transport thread will synchronously send all messages through the TCP socket. it will not drop messages after they are accepted to the buffer. 

in-memory transport simplifies this overflow behavior by replacing N sockets with buffer of 1000 with just one incoming `peer.socket` channel (not really a socket) with a buffer of N*1000. other than that the overflow policy to discard messages without reporting is similar between the two implementation